examples/Raytracer_Benchmark/RaytraceBenchmark.hpp

```diff
@@ -0,0 +1,147 @@
+    OPTION_INT(view, 1);
+
+    // zoom w.r.t to the default view mode
+    OPTION_DOUBLE(zoom, 3.5);
+
+    // Screen position in world coordinates
+    OPTION_DOUBLE(screenx, -5000);
+    OPTION_DOUBLE(screeny, 0);
+    OPTION_DOUBLE(screenz, 0);
+
+    // Up vector (no need to be normalized)
+    OPTION_DOUBLE(upx, 0);
+    OPTION_DOUBLE(upy, 1);
+    OPTION_DOUBLE(upz, 0);
+    vecgeom::Vector3D<double> up(upx, upy, upz);
+
+    // Light color, object color (no color per volume yet) - in RGBA chars compressed into an unsigned integer
+    OPTION_INT(bkgcol, 0xFF0000FF); // red
+    OPTION_INT(objcol, 0x0000FFFF); // blue
+    OPTION_INT(vdepth, 4);          // visible depth
+
+    copcore::Allocator<RaytracerData_t, backend> rayAlloc;
+    RaytracerData_t *rtdata = rayAlloc.allocate(1);
+  
+    rtdata->fScreenPos.Set(screenx, screeny, screenz);
+    rtdata->fUp.Set(upx, upy, upz);
+    rtdata->fZoom     = zoom;
+    rtdata->fModel    = (ERTmodel)model;
+    rtdata->fView     = (ERTView)view;
+    rtdata->fSize_px  = px;
+    rtdata->fSize_py  = py;
+    rtdata->fBkgColor = bkgcol;
+    rtdata->fObjColor = objcol;
+    rtdata->fVisDepth = vdepth;
+    rtdata->fMaxDepth = vecgeom::GeoManager::Instance().getMaxDepth();
+
+    Raytracer::InitializeModel((Raytracer::VPlacedVolumePtr_t)world, *rtdata);
+
+    if (backend == copcore::BackendType::CUDA) {
+
+  		cudaWorld((cuda::RaytracerData_t*)rtdata);
+
+    }
+    else {
+    	
+    	vecgeom::NavStateIndex vpstate;
+    	LoopNavigator::LocatePointIn(rtdata->fWorld, rtdata->fStart, vpstate, true);
+		rtdata->fVPstate = vpstate;
+
+    }
+
+    rtdata->Print();
+    
+    using RayBlock     = adept::BlockData<Ray_t>;
+    using RayAllocator = copcore::VariableSizeObjAllocator<RayBlock, backend>;
+    using Launcher_t   = copcore::Launcher<backend>;
+    using StreamStruct = copcore::StreamType<backend>;
+    using Stream_t     = typename StreamStruct::value_type;
+
+    // initialize BlockData of Ray_t structure
+    int capacity = 1 << 20;
+    RayAllocator hitAlloc(capacity);
+    RayBlock *rays = hitAlloc.allocate(1);
+
+    // Boilerplate to get the pointers to the device functions to be used
+    COPCORE_CALLABLE_DECLARE(generateFunc, generateRays);
+    COPCORE_CALLABLE_DECLARE(renderkernelFunc, renderKernels);
+
+    // Create a stream to work with.
+    Stream_t stream;
+    StreamStruct::CreateStream(stream);
+
+    // Allocate slots for the BlockData
+    Launcher_t generate(stream);
+    generate.Run(generateFunc, capacity, {0, 0}, rays);
+
+    generate.WaitStream();
+
+    // Allocate and initialize all rays on the host
+    size_t raysize = Ray_t::SizeOfInstance();
+    printf("=== Allocating %.3f MB of ray data on the %s\n", (float)rtdata->fNrays * raysize / 1048576, copcore::BackendName(backend));
+
+    copcore::Allocator<char, backend> charAlloc;
+    char *input_buffer = charAlloc.allocate(rtdata->fNrays*raysize);
+
+    copcore::Allocator<unsigned char, backend> ucharAlloc;
```

Comment:
Use NavIndex_t instead of unsigned char. This is defined in VecGeom/base/Global.h and it is a typedef to an unsigned int.

Reply:
So, should I change both char and unsigned char with NavIndex_t?

Reply:
Yes, and change the size accordingly